src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNet/DbCommandIntegration.cs

```diff
@@ -51,7 +56,7 @@ public static class DbCommandIntegration
                     MethodBuilder<Func<DbCommand, DbDataReader>>
                        .Start(moduleVersionPtr, mdToken, opCode, AdoNetConstants.MethodNames.ExecuteReader)
                        .WithConcreteType(typeof(DbCommand))
-                       .WithNamespaceAndNameFilters(DbDataReaderTypeName)
+                       .WithNamespaceAndNameFilters(AdoNetConstants.TypeNames.DbDataReader)
```

Comment:
This doesn't seem quite right for the `AdoNetConstants.MethodNames.ExecuteReaderExplicit` case. Wouldn't the return type `AdoNetConstants.TypeNames.DbDataReader` not match `AdoNetConstants.TypeNames.IDataReader` and cause `TryFindMethod()` to fail? Or are we not hitting that fallback logic?

Reply:
Good catch!

Reply:
One day I'm going to refactor `WithNamespaceAndNameFilters()` into `WithParameterTypes()` + `WithReturnType()`.

Reply:
Good idea. I always have to F12 to understand what it actually does